message_definitions/v1.0/ras_a.xml

```diff
@@ -10,7 +10,140 @@
+      </entry>
+      <entry value="40502" name="MAV_CMD_SET_EGRESS_PORTAL" hasLocation="true" isDestination="false">
+        <wip/>
+        <!-- This command is work-in-progress and it can therefore change. It should NOT be used in stable production environments. -->
+        <description>
+          Sets a specific portal to be an entry portal. Defines also the approach vector for a vehicle to approach and go through the portal.
+          COMMAND_INT should be used so to set the MAV_FRAME and consequently, frame and coordinates of the approach vector (MAV_FRAME_GLOBAL means global coordinates are being used,
+          while MAV_FRAME_LOCAL_NED means local NED coordinates are being used).
+        </description>
+        <param index="1" label="Portal ID" minValue="0" maxValue="4294967295" increment="1">ID of the portal to be set as an egress point.</param>
+        <param index="2" label="Approach vector initial point X or Latitude">X (mE4) or Latitude (degE7) (WGS84) coordinate of the initial point of the approach vector to the portal. NaN or INT32_MAX  if unknown.</param>
+        <param index="3" label="Approach vector initial point Y or Longitude">Y (mE4) or Longitude (degE7) (WGS84) coordinate of the initial point of the approach vector to the portal. NaN or INT32_MAX  if unknown.</param>
+        <param index="4" label="Approach vector initial point Z or Altitude (MSL)" units="m">Z or Altitude (MSL) coordinate of the initial point of the approach vector to the portal. NaN if unknown.</param>
+        <param index="5" label="Approach vector final point X or Latitude">X (mE4) or Latitude (degE7) (WGS84) coordinate of the final point of the approach vector to the portal. NaN or INT32_MAX  if unknown.</param>
+        <param index="6" label="Approach vector final point Y or Longitude">Y (mE4) or Longitude (degE7) (WGS84) coordinate of the final point of the approach vector to the portal. NaN or INT32_MAX if unknown.</param>
+        <param index="7" label="Approach vector final point Z or Altitude (MSL)" units="m">Z or Altitude (MSL) coordinate of the final point of the approach vector to the portal. NaN if unknown.</param>
+      </entry>
+      <entry value="40510" name="MAV_CMD_DO_EXPLORATION" hasLocation="true" isDestination="false">
+        <wip/>
+        <!-- This command is work-in-progress and it can therefore change. It should NOT be used in stable production environments. -->
+        <description>
+          Start or continue the exploration task.
+          If starting a new exploration, but requiring to get through a portal first, params 4 or 5 to 7 should be set and different from the specified ignore values.
+          Continuing an exploration can be done without necessarily setting the ingress portal, unless the exploration requires the definition of more than the initial ingress portal.
```

Comment:
I was about to ask why you need to define the portal position - why not just use the other messages above. It sounds like the case is multiple ingress points that must be navigated before entering a more normal exploration mode?

Reply:
It's useful to have an atomic enter-and-explore command, because building entry is a major cause of lost-link. If an entry is commanded separately, first, then the GCS might not be able to command the Explore.

Reply:
Assuming you can't know in advance if you'll lose the signal, doesn't that mean you always need to use the enter and explore? - i.e. if you use the enter, and then an expore you're in the same boat. 
So why have the separate enter command?

I'm sure this is bleedingly obvious, but since there are no workflows in the design doc ...

Reply:
Just completeness, I think. It's not unreasonable that a future craft might have an "enter portal" feature, but not an "exploration" feature, for example.

Reply:
Cool. Mostly I'm just whining because I don't feel I have the full picture, and that makes me uncomfortable.